tests/commands/envelope_test.py

```diff
@@ -355,3 +356,41 @@ def test_apply_from_user_and_email(self):
 
         self.assertTrue(env.sign)
         self.assertIs(env.sign_key, mock.sentinel.gpg_key)
+
+
+class TestSendCommand(unittest.TestCase):
+
+    """Tests for the SendCommand class."""
+
+    mail = textwrap.dedent("""\
+        From: foo@example.com
+        To: bar@example.com
+        Subject: FooBar
+
+        Foo Bar Baz
+        """)
+
+    def test_get_account_by_address_with_str(self):
+        cmd = envelope.SendCommand(mail=self.mail)
+        account = mock.Mock()
+        with mock.patch(
+                'alot.commands.envelope.settings.get_account_by_address',
+                mock.Mock(return_value=account)) as get_account_by_address:
+            cmd.apply(mock.Mock())
+        get_account_by_address.assert_called_once_with('foo@example.com',
+                                                       return_default=True)
+        # check that the apply did run through till the end.
+        account.send_mail.assert_called_once_with(self.mail)
+
+    def test_get_account_by_address_with_email_message(self):
+        mail = email.message_from_string(self.mail)
+        cmd = envelope.SendCommand(mail=mail)
+        account = mock.Mock()
+        with mock.patch(
+                'alot.commands.envelope.settings.get_account_by_address',
+                mock.Mock(return_value=account)) as get_account_by_address:
+            cmd.apply(mock.Mock())
+        get_account_by_address.assert_called_once_with('foo@example.com',
+                                                       return_default=True)
+        # check that the apply did run through till the end.
+        account.send_mail.assert_called_once_with(mail)
```

Comment:
I would rather use `account.send_mail.assert_called_once()` instead of dropping it entirely. I added this assert line in the first place because I had the feeling that I want to assert that one of the last lines of `apply()` is executed. Maybe I was just to confused by all these if branches that can `yield` and `return` and all these asynchronous things.

Proceed as you will.